chia/full_node/coin_store.py

```diff
@@ -130,6 +130,9 @@ async def get_coins_added_at_height(self, height: uint32) -> List[CoinRecord]:
         return coins
 
     async def get_coins_removed_at_height(self, height: uint32) -> List[CoinRecord]:
+        # Special case to avoid querying all unspent coins (spent_index=0)
+        if height == 0:
+            return []
         cursor = await self.coin_record_db.execute("SELECT * from coin_record WHERE spent_index=?", (height,))
```

Comment:
it seems this SQL query should be updated to say:
```
SELECT * from coin_record WHERE spent_index=? AND spent=1
```

Reply:
This was my first approach but I found it to still be slow for height 0 and actually slowed down non zero heights as well. May just have been a local issue though. It's definitely cleaner than the special case, so probably worth double checking

Reply:
it may be required to create a combined index of these two fields to make it fast. If there is an index for just `spent_index`, it should probably be extended to also include `spent`, since I suspect we always want to ask about both in combination (since `spent_index` doesn't have meaning for unspent coins).

In a future schema, `spent_index` could even be set to -1 to indicate that the coin is unspent, and save some space in the table.

Reply:
IIRC there is an index on `spent` - however, this actually can cause longer lookups. I think this is because since there are quite a few`spent=1` coins (and also of course a lot of `spent=0`), this causes sqlite to actually take longer than if you didn't use the index.

Reply:
There is an index on spent_height as well - but there is no combined index

Reply:
I agree with arvid that we should require spent =1.
Also if that is slow, we should fix that by using the spent_index as the DB index.

Anyways, I see nothing wrong with this PR